runtime/gc_base/IndexableObjectAllocationModel.cpp

```diff
@@ -251,6 +259,24 @@ MM_IndexableObjectAllocationModel::layoutDiscontiguousArraylet(MM_EnvironmentBas
 	if (NULL != spine) {
 		switch (_layout) {
 		case GC_ArrayletObjectModel::Discontiguous:
+#if defined(J9VM_GC_ENABLE_DOUBLE_MAP)
+			if (extensions->indexableObjectModel.isDoubleMappingEnabled() && extensions->isVLHGC()) {
```

Comment:
isVLHGC check is redundant, since isDoubleMappingEnabled will return true only in Balanced GC.